File: src/test/java/no/nav/dokgen/services/HateoasServiceTests.java
package no.nav.dokgen.services;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.test.context.junit4.SpringRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
public class HateoasServiceTests {

    @Before
    public void setUp() {

    }

    @Test
    public void skal_returnere_template_links() {
        var links = HateoasService.templateLinks("test");
        assertThat(links.getLinks().size()).isGreaterThan(0);
        for (var link : links.getLinks()) {
            assertThat(link.getRel()).isNotEmpty();
            assertThat(link.getHref()).isNotEmpty();
        }
    }

    @Test
    public void skal_returnere_testdata_links() {
        var links = HateoasService.testDataLinks("test", "test");
        assertThat(links.getLinks().size()).isGreaterThan(0);
        for (var link : links.getLinks()) {
            assertThat(link.getRel()).isNotEmpty();
            assertThat(link.getHref()).isNotEmpty();
        }
    }
}

File: src/test/kotlin/no/nav/dokgen/services/HateoasServiceTests.kt
package no.nav.dokgen.services

import no.nav.dokgen.services.HateoasService.templateLinks
import no.nav.dokgen.services.HateoasService.testDataLinks
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(SpringExtension::class)
class HateoasServiceTests {

    @Test
    fun skal_returnere_template_links() {
        val links = templateLinks("test")
        Assertions.assertThat(links.links).isNotEmpty
        for (link in links.links) {
            Assertions.assertThat(link.rel).isNotNull
            Assertions.assertThat(link.href).isNotEmpty
        }
    }

    @Test
    fun skal_returnere_testdata_links() {
        val links = testDataLinks("test", "test")
        Assertions.assertThat(links.links).isNotEmpty
        for (link in links.links) {
            Assertions.assertThat(link.rel).isNotNull
            Assertions.assertThat(link.href).isNotEmpty
        }
    }
}